lib/Service/AuthenticationService.php

```diff
@@ -136,5 +165,99 @@ public function fetchOAuthTokens (string $endpoint, array $configuration): strin
 
 		return $result['access_token'];
 	}
+	private function getRSJWK(array $configuration): ?JWK
+	{
+		$stamp = microtime().getmypid();
+		file_put_contents("/srv/api/var/privatekey-$stamp", $configuration['secret']);
+		$jwk = null;
+		$filename = "/srv/api/var/privatekey-$stamp";
+		try{
+			$jwk = JWKFactory::createFromKeyFile(
+				$filename,
+				null,
+				['use' => 'sig']
+			);
+		}
+		catch(\Exception $exception) {
+
+		}
+
+		unlink($filename);
+
+		return $jwk;
+	}
+
+	private function getHSJWK(array $configuration): ?JWK
+	{
+		return new JWK(
+			[
+				'kty' => 'oct',
+				'k'   => rtrim(string: base64_encode(addslashes($configuration['secret'])), characters: '='),
+			]
+		);
+	}
+
+	private function getJWTPayload(array $configuration): array
+	{
+		$renderedPayload = $this->twig->createTemplate($configuration['payload'])->render($configuration);
+
+		return json_decode($renderedPayload, true);
+	}
+
+	private function getJWK(array $configuration): ?JWK
+	{
+		$jwk = null;
+		if (in_array(needle: $configuration['algorithm'], haystack: ['HS256', 'HS512'])) {
+			$jwk = $this->getHSJWK($configuration);
+		} else if (in_array(needle: $configuration['algorithm'], haystack: ['RS256', 'RS384', 'RS512'])) {
```

Comment:
avoid else if, just use return

Reply:
else if is needed here because the default should be that the algorithm is not supported